### ISSUE TITLE
Fix auth function code error which causes bcsc login failure 

### ISSUE THREAD

**Issue by MCatherine1994** [bcgov/nr-forests-access-management#1017]
**Describe the task**
Forest client can't login using bc service card credential, and by testing using debugger tool and the cloud watch log, found a python key error in the auth function when try to write the audit log. But we still need to verify if this is the real problem, or the return data from the bcsc login changed. 

<img width="1243" alt="Screen Shot 2023-11-02 at 1 20 30 PM" src="https://github.com/bcgov/nr-forests-access-management/assets/23131735/011fb33b-0eb0-4bf5-ba97-5eecab24a069">

<img width="1427" alt="Screen Shot 2023-11-02 at 1 23 02 PM" src="https://github.com/bcgov/nr-forests-access-management/assets/23131735/199092aa-84b2-453f-baa8-69caa1c3e486">

**Acceptance Criteria**

- [x] Verify the cause of the problem
- [x] If it's the python key error, fix it

**Additional context**
- Add any other context about the task here.
- Or here

**Comment by ianliuwk1019** [comment]
Added some finding for further investigation. According to BCSC mapping table, there is no "**custom:idp_username**" .
[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMitnQVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--e99ae85ad00684eb58a3692ef4f43c05cfcab02e/image.png)

**Comment by MCatherine1994** [comment]
Found a weird pipeline issue that is not related to this ticket, but happened when try to do dev deployment this afternoon. Just write this down in case we see the similar issue in the future. The issue can be found [here](https://github.com/bcgov/nr-forests-access-management/actions/runs/6749956121/job/18351916145). 

<img width="1130" alt="Screen Shot 2023-11-03 at 4 29 39 PM" src="https://github.com/bcgov/nr-forests-access-management/assets/23131735/6a4c7463-f581-421e-8e51-0fa873b9c4a7">

Basically it complains the state lock file in the Dynamodb. Ian and I compared the state items in our AWS dev environment and AWS test environment. We found that somehow dev has an extra entry which we don't know why and how it got created. The issue was solved by remove that extra entry manually and rerun the pipeline.
![image](https://github.com/bcgov/nr-forests-access-management/assets/23131735/e2c0b6c6-2df2-4565-b0ff-4f68db44cc60)

**Comment by MCatherine1994** [comment]
We also found another issue for the bcsc login, also not related to the issue in this ticket. 

When try the BCSC login (using test user account) using the debugger tool with our AWS DEV Cognito endpoint and TEST Cognito endpoint, they both return a username attribute mapping error and failed to return the authentication code. 

![image](https://github.com/bcgov/nr-forests-access-management/assets/23131735/cb130a84-c3d9-4729-adbc-ae6e32d87fc2)

But if try the BCSC login with our AWS PROD Cognito endpoint, it works fine at this step, I can get the authentication code back. So our three AWS environment (dev, test, prod) is not consistent. 


And TOOLS space is different than the rest. When try the BCSC and BCEID login using the debugger tool with our AWS TEST Cognito endpoint, I got a redirect url error. Maybe we didn't setup the tools space to support them. IDIR login works fine.
![image](https://github.com/bcgov/nr-forests-access-management/assets/23131735/952733f2-a601-4869-8c40-97924d2532f5)

**Comment by MCatherine1994** [comment]
Because our AWS dev space doesn't support BCSC login, I couldn't confirm what data we can get in the auth lambda function for bcsc user. Will only add the unit test for bceid user for now. The scope for this ticket is to solve the bcsc login problem in prod for team Alliance. Once we confirm the return data in our prod space, will create another ticket to add the unit test for bcsc user.
